versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/extension/TestConnectionProviderSource.java

```diff
@@ -91,15 +91,7 @@ boolean isCompatibleWith(
    */
   DatabaseConnectionProvider<CONN_CONFIG> createConnectionProvider();
 
-  /**
-   * Updates the given {@code config} with the {@link DatabaseConnectionProvider} via {@link
-   * DatabaseAdapterConfig#withConnectionProvider(DatabaseConnectionProvider)}.
-   *
-   * @param config the current configuration
-   * @return {@code config} with a valid {@link DatabaseConnectionProvider} instance.
-   */
-  DatabaseAdapterConfig<DatabaseConnectionProvider<CONN_CONFIG>> updateConfig(
-      DatabaseAdapterConfig<DatabaseConnectionProvider<CONN_CONFIG>> config);
+  DatabaseConnectionProvider<CONN_CONFIG> getConnectionProvider();
```

Comment:
A javadoc for this method would be nice - and outline the difference of `createConnectionProvider` vs `getConnectionProvider`

Reply:
I'm not clear on this myself... perhaps we do not really need `createConnectionProvider` now :thinking:

Reply:
I moved `createConnectionProvider` to the impl. class in 1b903edb794652ba6259d3878ad87d47eeaa3a37... hope it's clearer now.... also update javadoc.